src/openms/include/OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h

```diff
@@ -0,0 +1,173 @@
+
+    /**
+      @brief Run the full analysis for the experiment for the given time interval
+
+      The workflow steps are:
+      - the time axis of the experiment is cut to n_seconds
+      - the spectra are summed into one along the time axis with the bin size determined by mz and instrument resolution
+      - data is smoothed by applying the Savitzky-Golay filter
+      - peaks are picked
+      - the accurate mass search for all the picked peaks is performed
+
+      The intermediate summed spectra and picked peaks can be saved to the filesystem. 
+      Also, the results of the accurate mass search and the signal-to-noise information 
+      of the resulting spectrum is saved.
+
+      @param experiment  Input MSExperiment
+      @param n_seconds Input number of seconds
+      @param load_cached_spectrum Load the cached picked spectrum if exists
+      @param output   [out] Output of the accurate mass search results
+      @return a boolean indicating if the picked spectrum was loaded from the cached file
+    */
+    bool run(const MSExperiment & experiment, const float & n_seconds, OpenMS::MzTab & output, const bool load_cached_spectrum = true);
+
+    /**
+      @brief Cut the time axis of the experiment up to @n_seconds
+
+      @param experiment  Input MSExperiment
+      @param n_seconds Input number of seconds
+      @param output   [out] Spectra with retention time less than @n_seconds
+    */
+    void cutForTime(const MSExperiment & experiment, const float & n_seconds, std::vector<MSSpectrum> & output);
+
+    /**
+      @brief Sum the spectra with different retention times into one.
+
+      The bin size for summing the intensities is defined as mz / (resolution*4) 
+      for all the mzs taken with the @bin_step defined in the parameters
+
+      @param input  Input vector of spectra
+      @return a spectrum
+    */
+    MSSpectrum mergeAlongTime(const std::vector<OpenMS::MSSpectrum> & input);
+
+    /**
+      @brief Pick peaks from the summed spectrum
+
+      @param input  Input vector of spectra
+      @return a spectrum with picked peaks
+    */
+    MSSpectrum extractPeaks(const MSSpectrum & input);
+
+    /**
+      @brief Convert a spectrum to a feature map with the corresponding polarity
+
+      @param input  Input a picked spectrum
+      @return a feature map with the peaks converted to features and polarity from the parameters
+    */
+    FeatureMap convertToFeatureMap(const MSSpectrum & input);
+
+    /**
+      @brief Estimate noise for each peak
+
+      @param input  Input a picked spectrum
+      @return a spectrum object storing logSN information
+    */
+    MSSpectrum trackNoise(const MSSpectrum & input);
+
+    /**
+      @brief Perform accurate mass search
+
+      @param input  Input a feature map
+      @param output  [out] mzTab file with the accurate mass search results
+    */
+    void runAccurateMassSearch(FeatureMap & input, OpenMS::MzTab & output);
```

Comment:
coding style: please put the ampersand next to the type, e.g. `FeatureMap& input`